packages/javascript/bh-shared-ui/src/components/CreateUserForm/CreateUserForm.tsx

```diff
@@ -14,426 +14,696 @@
 import { apiClient } from '../../utils';
 
-export type CreateUserRequestForm = Omit<CreateUserRequest, 'SSOProviderId'> & { SSOProviderId: string | undefined };
+export type CreateUserRequestForm = Omit<CreateUserRequest, 'sso_provider_id'> & {
+    sso_provider_id: string | undefined;
+};
 
 const CreateUserForm: React.FC<{
-    onCancel: () => void;
-    onSubmit: (user: CreateUserRequestForm) => void;
-    isLoading: boolean;
     error: any;
-}> = ({ onCancel, onSubmit, isLoading, error }) => {
-    const {
-        control,
-        handleSubmit,
-        setValue,
-        formState: { errors },
-        setError,
-    } = useForm<CreateUserRequestForm>({
-        defaultValues: {
-            emailAddress: '',
-            principal: '',
-            firstName: '',
-            lastName: '',
-            password: '',
-            needsPasswordReset: false,
-            roles: [1],
-            SSOProviderId: '',
+    isLoading: boolean;
+    onSubmit: (user: CreateUserRequestForm) => void;
+    showEnvironmentAccessControls?: boolean;
+}> = ({ error, isLoading, onSubmit, showEnvironmentAccessControls }) => {
+    const defaultValues = {
+        email_address: '',
+        principal: '',
+        first_name: '',
+        last_name: '',
+        password: '',
+        needs_password_reset: false,
+        roles: [3],
+        sso_provider_id: '',
+        all_environments: false,
+        environment_targeted_access_control: {
+            environments: null,
         },
-    });
+    };
+
+    const form = useForm<CreateUserRequestForm>({ defaultValues });
 
     const [authenticationMethod, setAuthenticationMethod] = React.useState<string>('password');
+    const [needsPasswordReset, setNeedsPasswordReset] = useState(false);
+    const [searchInput, setSearchInput] = useState<string>('');
+    const [selectedEnvironments, setSelectedEnvironments] = useState<string[]>([]);
+    const [selectedRoleValue, setSelectedRoleValue] = useState([3]);
 
-    useEffect(() => {
-        if (authenticationMethod === 'password') {
-            setValue('SSOProviderId', undefined);
+    const handleCheckedChange = (checked: boolean | 'indeterminate') => {
+        setNeedsPasswordReset(checked === true);
+        if (checked === true) {
+            form.setValue('needs_password_reset', true);
+        } else {
+            form.setValue('needs_password_reset', false);
+        }
+    };
+
+    const getRolesQuery = useQuery(['getRoles'], ({ signal }) =>
+        apiClient.getRoles({ signal }).then((res) => res.data?.data?.roles)
+    );
+
+    const listSSOProvidersQuery = useQuery(['listSSOProviders'], ({ signal }) =>
+        apiClient.listSSOProviders({ signal }).then((res) => res.data?.data)
+    );
+
+    const roleInputValue = form.watch('roles');
+    const selectedETACEnabledRole = roleInputValue.toString() === '2' || roleInputValue.toString() === '3';
```

Comment:
Is this the only place we need to know if a user role is an ETAC enabled role? If not, id suggest we pull out this into a function like `isUserRoleETACEnabled(roleInputValue.toString())` so we can reuse this logic in the future and if a user role becomes ETAC enabled then we can easily add it to the check.

Reply:
Im seeing the chonky CR comment about this and had similar feelings but figured it wasnt much of a risk right not considering those IDs are in the main migration schema. 

If we did want to address this, I think the easiest/safest way is instead of a `isUserRoleETACEnabled` util, we add a hook that pulls in the `useListDisplayRoles` and returns two functions, one that checks if a role id is ETAC enabled, or that a role name is ETAC enabled! And then at this point it would make sense to encode what role names are considered ETAC enabled.

Im happy to help figure out where and what these types+functions should be!